js/lib/ui.widgets.js

```diff
@@ -1,211 +1,15 @@
-	if (multiple === false) {
-		$button.addClass('elgg-state-available');
-		$button.removeClass('elgg-state-unavailable');
-		$button.unbind('click', elgg.ui.widgets.add); // make sure we don't bind twice
-		$button.click(elgg.ui.widgets.add);
-	}
-
-	$widget.remove();
-
-	// delete the widget through ajax
-	elgg.action($(this).attr('href'));
-
-	event.preventDefault();
-};
-
-/**
- * Toggle the collapse state of the widget
- *
- * @param {Object} event
- * @return void
- */
-elgg.ui.widgets.collapseToggle = function(event) {
-	$(this).toggleClass('elgg-widget-collapsed');
-	$(this).parent().parent().find('.elgg-body').slideToggle('medium');
-	event.preventDefault();
-};
-
-/**
- * Save a widget's settings
- *
- * Uses Ajax to save the settings and updates the HTML.
- *
- * @param {Object} event
- * @return void
- */
-elgg.ui.widgets.saveSettings = function(event) {
-	$(this).parent().slideToggle('medium');
-	var $widgetContent = $(this).parent().parent().children('.elgg-widget-content');
-
-	// stick the ajax loader in there
-	var $loader = $('#elgg-widget-loader').clone();
-	$loader.attr('id', '#elgg-widget-active-loader');
-	$loader.removeClass('hidden');
-	$widgetContent.html($loader);
-
-	var default_widgets = $("input[name='default_widgets']").val() || 0;
-	if (default_widgets) {
-		$(this).append('<input type="hidden" name="default_widgets" value="1">');
-	}
-
-	elgg.action('widgets/save', {
-		data: $(this).serialize(),
-		success: function(json) {
-			$widgetContent.html(json.output);
-			if (typeof(json.title) != "undefined") {
-				var $widgetTitle = $widgetContent.parent().parent().find('.elgg-widget-title');
-				$widgetTitle.html(json.title);
-			}
-		}
-	});
-	event.preventDefault();
-};
-
-/**
- * Set the min-height so that all widget column bottoms are the same
- *
- * This addresses the issue of trying to drag a widget into a column that does
- * not have any widgets or many fewer widgets than other columns.
- *
- * @param {String} selector
- * @return void
- */
-elgg.ui.widgets.setMinHeight = function(selector) {
-	var maxBottom = 0;
-	$(selector).each(function() {
-		var bottom = parseInt($(this).offset().top + $(this).height());
-		if (bottom > maxBottom) {
-			maxBottom = bottom;
-		}
-	});
-	$(selector).each(function() {
-		var bottom = parseInt($(this).offset().top + $(this).height());
-		if (bottom < maxBottom) {
-			var newMinHeight = parseInt($(this).height() + (maxBottom - bottom));
-			$(this).css('min-height', newMinHeight + 'px');
-		}
-	});
-};
-
-elgg.register_hook_handler('init', 'system', elgg.ui.widgets.init);
+elgg.ui.widgets = function() {
+	elgg.deprecated_notice("elgg.ui.widgets object has been converted to 'elgg/widgets' AMD module", '2.1');
+}
+
+elgg.ui.widgets = {
+	init: elgg.ui.widgets,
+	add: elgg.ui.widgets,
+	move: elgg.ui.widgets,
+	remove: elgg.ui.widgets,
+	collapseToggle: elgg.ui.widgets,
+	saveSettings: elgg.ui.widgets,
+	setMinHeight: elgg.ui.widgets,
+};
```

Comment:
It could if we are doing it just for the sake of creating an AMD module. I don't really a see a point though, as it will still be loaded on each page...